### ISSUE TITLE
Snap to grid only working for points

### ISSUE THREAD

**Issue by elkins0** [CindyJS/CindyJS#80]
In this line through point CindyJS example:

```javascript
createCindy({
	scripts: "cs*", 
	defaultAppearance: { dimDependent: 0.7 },
	grid: 1,
	snap: true,
	geometry: [ 
		{ name: "A", type: "Free", pos: [ -2.0317460317460316, -4.0, -0.7936507936507936 ], color: [ 1.0, 0.0, 0.0 ], labeled: true }, 
		{ name: "a", type: "Through", pos: [ -0.3625983644499306, -0.6094738466286066, 4.0 ], color: [ 0.0, 0.0, 1.0 ], args: [ "A" ], labeled: true, size: 1 } ], 
	ports: [ 
		{ id: "CSCanvas", width: 680, height: 350, transform: [ { visibleRect: [ -9.04, 9.32, 18.16, -4.68 ] } ], background: "rgb(168,176,192)" } ] });
```

Moving the point A near the grid snaps the point to the grid. Moving the angle of line does not.

Instead of placing the same snapping functionality inside found inside `geoOps.Free.getParamForInput` into every `geoOps.*.getParamForInput` perhaps it could be placed inside traceMover.

In https://github.com/CindyJS/CindyJS/tree/master/src/js/libgeo/Tracing.js:
```javascript
function traceMover(mover, pos, type) {
...
    // snap pos to grid before making the call to getParamForInput
    var targetParam = opMover.getParamForInput(mover, pos, type);
```

**Comment by gagern** [comment]
I don't think we should do the snapping in `traceMover`. The main reason are semi-free elements. When moving a point on a line, then snapping the point to grid and then projecting it to the line seem wrong. I'm not sure about right behavior here, but I'd probably expect projecting the point onto the line first, and then snapping it to the nearest intersection of the line and the grid lines. Except if the line coincides with a grid line, in which case we only consider the orthogonal portion of the grid.

So we need to have snapping depend on the operation in question. We still should snap the `Through` line to grid coordinates, though. Having that functionality as a helper which can be reused from different operations should be a good solution I think.
